fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/QueryToKeyMatcher.java

```diff
@@ -223,8 +223,52 @@ public Match matches(@Nonnull KeyExpression expression) {
     }
 
     @Nonnull
-    private Match matches(@Nonnull AndComponent query, @Nonnull KeyExpression key, @Nonnull MatchingMode matchingMode,
-                          @Nullable FilterSatisfiedMask filterMask) {
+    private Match matches(@Nonnull QueryComponent query, @Nonnull KeyExpression key,
```

Comment:
The first two overloaded methods were swapped because the first one is the actual (after setup) entry point and recursion point.